code/MimeUploadValidator.php

```diff
@@ -88,11 +88,11 @@ public function getExpectedMimeTypes($tmpFile)
         // Get the mime types and their variations from mimevalidator
         $knownMimes = Config::inst()->get(get_class($this), 'MimeTypes');
         if (isset($knownMimes[$extension])) {
-            if (is_array($knownMimes[$extension])) {
```

Comment:
You've broken non-array mime type config. You need to retain the is_array() check.